echo-telemetry/src/main/java/com/netflix/spinnaker/echo/telemetry/InstanceIdSupplier.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 Google, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.telemetry
+
+import com.netflix.spinnaker.echo.config.TelemetryConfig
+import com.netflix.spinnaker.kork.jedis.RedisClientSelector
+import com.netflix.spinnaker.kork.jedis.exception.RedisClientNotFound
+import java.lang.RuntimeException
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+import redis.clients.jedis.commands.JedisCommands
+
+val UNIQUE_ID_KEY = "spinnaker:stats:instanceId"
+
+@Component
+@ConditionalOnProperty("stats.enabled")
+class InstanceIdSupplier(
+  private val config: TelemetryConfig.TelemetryConfigProps,
+  redisSelector: RedisClientSelector?
+) {
+  private val log: Logger = LoggerFactory.getLogger(this.javaClass)
+  private val redis = try { redisSelector?.primary("default") } catch (e: RedisClientNotFound) { null }
+  val uniqueId = getOrSetId(config.instanceId)
```

Comment:
This does blocking IO during object construction, which is frowned upon. You can do this, which will run the method the first time `uniqueId` is referenced in a client:
`val uniqueId by lazy { getOrSetId(config.instanceId) }`
It's a little weird since callers have no way of knowing that accessing `uniqueId` might do I/O, but since it's only really used one place, I think it's probably fine.

Alternately, you could use a Spring `@PostConstruct` annotation, but then `uniqueId` has to be a `var String?` instead of `val String`, which is annoying...

Reply:
I had this as `lazy` initially, then thought maybe that solution was overly-complex so reverted back to regular initialization.  I'll replace it with `lazy`!